datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -933,24 +965,59 @@ impl ExecutionPlan for SortExec {
     }
 
     fn with_fetch(&self, limit: Option<usize>) -> Option<Arc<dyn ExecutionPlan>> {
-        Some(Arc::new(SortExec {
-            input: Arc::clone(&self.input),
-            expr: self.expr.clone(),
-            metrics_set: self.metrics_set.clone(),
-            preserve_partitioning: self.preserve_partitioning,
-            fetch: limit,
-            cache: self.cache.clone(),
-        }))
+        Some(Arc::new(SortExec::with_fetch(self, limit)))
     }
 
     fn fetch(&self) -> Option<usize> {
         self.fetch
     }
 }
 
+struct TopKStream {
+    input: SendableRecordBatchStream,
+    schema: SchemaRef,
+    fetch: usize,
+}
+
+impl Stream for TopKStream {
```

Comment:
This looks very similar to `LimitStream` -- https://docs.rs/datafusion-physical-plan/41.0.0/src/datafusion_physical_plan/limit.rs.html#434

though limit stream has metrics and some other features

Reply:
Yes, maybe we can reuse it. We will take a look

Reply:
You are right -- sent a commit to reuse `LimitStream`